### PR TITLE
Handling TO_TIMESTAMP with different operands and time formats.

### PR DESCRIPTION
**Pull request by nirnaypatil11**

### CODE REVIEW

**Review by nosheenSheikh02 — COMMENTED** [review]
Add test cases

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 1]
```diff
@@ -1187,8 +1187,10 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
           ? ((NlsString) requireNonNull(((SqlCharStringLiteral) call.operand(0)).getValue()))
```
[the remark this answers is not in the document]
Comment (nirnaypatil11) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1187]: resolved

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 2]
```diff
@@ -1187,8 +1187,10 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
...
           .getValue()
           : call.operand(0).toString();
+      String finalDateFormat = dateFormt.replaceAll("S\\(\\d\\)",
+          SqlDateTimeFormat.MILLISECONDS_5.value);
       SqlCall formtCall = PARSE_TIMESTAMP.createCall(SqlParserPos.ZERO,
-          createDateTimeFormatSqlCharLiteral(dateFormt), call.operand(1));
+          createDateTimeFormatSqlCharLiteral(finalDateFormat), call.operand(1));
```
[the remark this answers is not in the document]
Comment (nirnaypatil11) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1193]: resolved

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 3]
```diff
@@ -1352,13 +1354,31 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
...
       break;
     case "TIMESTAMP_MICROS":
-      castAsDatetime(writer, call, leftPrec, rightPrec, TIMESTAMP_MICROS);
+      if (call.operandCount() == 1) {
+        castAsDatetime(writer, call, leftPrec, rightPrec, TIMESTAMP_MICROS);
+      } else {
+        SqlCall timeStampMicroCall = TIMESTAMP_MICROS.createCall(SqlParserPos.ZERO,
+            new SqlNode[] { call.operand(0) });
+        TIMESTAMP_MICROS.unparse(writer, timeStampMicroCall, leftPrec, rightPrec);
+      }
```
[the remark this answers is not in the document]
Comment (nirnaypatil11) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1381]: resolved

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (nirnaypatil11) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java]: done

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 4]
```diff
@@ -13596,4 +13616,17 @@ private RexNode createRexForQuantile(RelBuilder builder) {
...
+//        + " PERCENTILE_CONT(0.25) WITHIN GROUP (ORDER BY \"product_id\")\n"
+//        + "FROM \"product\"";
+//    final String expectedSql = "SELECT PERCENTILE_CONT(0.25) WITHIN GROUP "
+//        + "(ORDER BY \"product_id\")\n"
+//        + "FROM \"foodmart\".\"product\"";
+//
+//    sql(query)
+//        .ok(expectedSql);
+//
+//  }
```
[the remark this answers is not in the document]
Comment (nirnaypatil11) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: resolved

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 5]
```diff
@@ -1681,6 +1675,17 @@ private void unparseRegexMatchCount(SqlWriter writer, SqlCall call,
     writer.endFunCall(arrayLengthFrame);
   }
 
+  private void unParseTimestampSeconds(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec,
+      SqlFunction sqlFunction) {
+    if (call.operandCount() == 1) {
+      castAsDatetime(writer, call, leftPrec, rightPrec, sqlFunction);
+    } else {
+      SqlCall timestampSecondsCall = sqlFunction.createCall(SqlParserPos.ZERO,
```
Comment (dm-shantanuc) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1683]: This behavior should only be there in case second operand is true.

Comment (nirnaypatil11) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1681]: resolved

**Review by dm-shantanuc — APPROVED** [review]
LGTM!

**Pull request merged**